src/dft.cpp

```diff
@@ -1063,13 +979,27 @@ void fields::output_dft_components(dft_chunk **chunklists, int num_chunklists,
    if (chunklists[nc])
     NumFreqs = chunklists[nc]->Nomega;
 
+  // if the volume has zero thickness in one or more directions, the DFT
+  // grid is two pixels thick in those directions, but we want the HDF5 output
+  // to be just one pixel thick in those directions. solution: first get the
+  // fields in array form (as get_dft_array), then collapse degenerate dimensions
+  // and export the collapsed array to HDF5. in this case the max_to_all() below
+  // is needed to make sure everybody agrees on how many frequencies there are,
+  // because some processes' field chunks may have no overlap with dft_volume,
+  // in which case those processes will think NumFreqs==0.
   bool have_empty_dims=false;
   LOOP_OVER_DIRECTIONS(dft_volume.dim, d)
    if (dft_volume.in_direction(d)==0.0) 
     have_empty_dims=true;
 
-  //h5file *f = (have_empty_dims && am_master()) ? open_h5file(HDF5FileName,h5file::WRITE) : 0;
-  h5file::access_mode mode = h5file::WRITE;
+  h5file *file=0;
+  if ( have_empty_dims && am_master() )
+   { char filename[100];
+     snprintf(filename,100,"%s%s",HDF5FileName,strstr("%.h5",HDF5FileName) ? "" : ".h5");
+     file = new h5file(filename,h5file::WRITE, false /*parallel*/ );
+   }
+  if (have_empty_dims)
+   NumFreqs = (int)max_to_all( (double)NumFreqs ); // subtle!
```

Comment:
Note that there is a `int max_to_all(int in)` method

Reply:
My mistake. This call started out as a call to `max_to_master`, for which there doesn't appear to be an `int`-valued version. For micro-revisions like this, in the interest of speeding the pace of development I really encourage you to go ahead and make the revisions yourself.